clientfactory/encryptionclient.go

```diff
@@ -0,0 +1,13 @@
+package clientfactory
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3crypto"
+)
+
+//NewEncryptionClient returns an s3crypto encryption client
+func NewEncryptionClient(sess *session.Session, cipher s3crypto.ContentCipherBuilder) *s3crypto.EncryptionClient {
```

Comment:
I think my other comment was removed when the first was removed - so just reposting:

Is there a benefit to this being outside the put file? It just seems that it takes the arguments and passes them to the AWS API. So if there aren't multiple steps which this is combining, I don't see the net gain. Unless it helps with mocking/testing?

Reply:
I was going for a couple things, allows us to follow the mock testing pattern, and I was trying to clean up the function that performs the put. If you look back at earlier iterations of the function there is a lot going on in there.

Reply:
I know there was a lot going on - but this doesn't reduce the amount going on. In the end, the lines of code doesn't actually change, right?